xarray/tests/test_plot.py

```diff
@@ -1794,6 +1800,121 @@ def test_origin_overrides_xyincrease(self):
             assert plt.ylim()[0] < 0
 
 
+# The try/except/else is needed for the matplotlib version check, to handle the case
+# when matplotlib is not installed. It should be possible to remove it once we require
+# matplotlib>=3.2.0.
+# Note, importing as mpl2 to avoid redefining mpl, which is a flake8 error.
+try:
+    import matplotlib as mpl2
+except ImportError:
+    pass
+else:
+
+    @pytest.mark.skipif(
+        LooseVersion(mpl2.__version__) < "3.2.0",
+        reason="surface plot requires newer matplotlib",
+    )
+    class TestSurface(Common2dMixin, PlotTestCase):
+
+        plotfunc = staticmethod(xplt.surface)
+        subplot_kws = {"projection": "3d"}
+
+        def test_primitive_artist_returned(self):
+            artist = self.plotmethod()
+            assert isinstance(artist, mpl_toolkits.mplot3d.art3d.Poly3DCollection)
+
+        @pytest.mark.slow
+        def test_2d_coord_names(self):
+            self.plotmethod(x="x2d", y="y2d")
+            # make sure labels came out ok
+            ax = plt.gca()
+            assert "x2d" == ax.get_xlabel()
+            assert "y2d" == ax.get_ylabel()
+            assert f"{self.darray.long_name} [{self.darray.units}]" == ax.get_zlabel()
+
+        def test_xyincrease_false_changes_axes(self):
+            # Does not make sense for surface plots
+            pass
+
+        def test_xyincrease_true_changes_axes(self):
+            # Does not make sense for surface plots
+            pass
+
+        def test_can_pass_in_axis(self):
+            self.pass_in_axis(self.plotmethod, subplot_kw={"projection": "3d"})
+
+        def test_default_cmap(self):
+            # Does not make sense for surface plots with default arguments
+            pass
+
+        def test_diverging_color_limits(self):
+            # Does not make sense for surface plots with default arguments
+            pass
+
+        def test_colorbar_kwargs(self):
+            # Does not make sense for surface plots with default arguments
+            pass
+
+        def test_cmap_and_color_both(self):
+            # Does not make sense for surface plots with default arguments
+            pass
+
+        def test_seaborn_palette_as_cmap(self):
+            # seaborn does not work with mpl_toolkits.mplot3d
```

Comment:
Why not? I would not expect an interaction here? Or is it because you don't actually call the code determining the cmap?

Reply:
I don't understand the error. If I don't modify the test, I get
```
================================================================================================================== FAILURES ==================================================================================================================
__________________________________________________________________________________________________ TestSurface.test_seaborn_palette_as_cmap __________________________________________________________________________________________________

self = <xarray.tests.test_plot.TestSurface object at 0x7f5043826220>

    @requires_seaborn
    def test_seaborn_palette_as_cmap(self):
>       cmap_name = self.plotmethod(levels=2, cmap="husl").get_cmap().name

.../xarray/xarray/tests/test_plot.py:1232: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.../xarray/xarray/plot/plot.py:873: in plotmethod
    return newplotfunc(**allargs)
.../xarray/xarray/plot/plot.py:775: in newplotfunc
    primitive = plotfunc(
.../xarray/xarray/plot/plot.py:1035: in surface
    primitive = ax.plot_surface(x, y, z, **kwargs)
.../python3.8/site-packages/matplotlib/_api/deprecation.py:431: in wrapper
    return func(*inner_args, **inner_kwargs)
.../python3.8/site-packages/mpl_toolkits/mplot3d/axes3d.py:1742: in plot_surface
    polyc = art3d.Poly3DCollection(polys, *args, **kwargs)
.../python3.8/site-packages/mpl_toolkits/mplot3d/art3d.py:715: in __init__
    super().__init__(verts, *args, **kwargs)
.../python3.8/site-packages/matplotlib/collections.py:1206: in __init__
    super().__init__(**kwargs)
.../python3.8/site-packages/matplotlib/_api/deprecation.py:431: in wrapper
    return func(*inner_args, **inner_kwargs)
.../python3.8/site-packages/matplotlib/collections.py:164: in __init__
    cm.ScalarMappable.__init__(self, norm, cmap)
.../python3.8/site-packages/matplotlib/cm.py:263: in __init__
    self.set_cmap(cmap)  # The Colormap instance of this ScalarMappable.
.../python3.8/site-packages/matplotlib/cm.py:432: in set_cmap
    cmap = get_cmap(cmap)
.../python3.8/site-packages/matplotlib/cm.py:190: in get_cmap
    _api.check_in_list(sorted(_cmap_registry), name=name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

_values = ['Accent', 'Accent_r', 'Blues', 'Blues_r', 'BrBG', 'BrBG_r', ...], _print_supported_values = True, kwargs = {'name': 'husl'}, values = ['Accent', 'Accent_r', 'Blues', 'Blues_r', 'BrBG', 'BrBG_r', ...], key = 'name', val = 'husl'

    def check_in_list(_values, *, _print_supported_values=True, **kwargs):
        """
        For each *key, value* pair in *kwargs*, check that *value* is in *_values*.
    
        Parameters
        ----------
        _values : iterable
            Sequence of values to check on.
        _print_supported_values : bool, default: True
            Whether to print *_values* when raising ValueError.
        **kwargs : dict
            *key, value* pairs as keyword arguments to find in *_values*.
    
        Raises
        ------
        ValueError
            If any *value* in *kwargs* is not found in *_values*.
    
        Examples
        --------
        >>> _api.check_in_list(["foo", "bar"], arg=arg, other_arg=other_arg)
        """
        values = _values
        for key, val in kwargs.items():
            if val not in values:
                if _print_supported_values:
>                   raise ValueError(
                        f"{val!r} is not a valid value for {key}; "
                        f"supported values are {', '.join(map(repr, values))}")
E                   ValueError: 'husl' is not a valid value for name; supported values are 'Accent', 'Accent_r', 'Blues', 'Blues_r', 'BrBG', 'BrBG_r', 'BuGn', 'BuGn_r', 'BuPu', 'BuPu_r', 'CMRmap', 'CMRmap_r', 'Dark2', 'Dark2_r', 'GnBu', 'GnBu_r', 'Greens', 'Greens_r', 'Greys', 'Greys_r', 'OrRd', 'OrRd_r', 'Oranges', 'Oranges_r', 'PRGn', 'PRGn_r', 'Paired', 'Paired_r', 'Pastel1', 'Pastel1_r', 'Pastel2', 'Pastel2_r', 'PiYG', 'PiYG_r', 'PuBu', 'PuBuGn', 'PuBuGn_r', 'PuBu_r', 'PuOr', 'PuOr_r', 'PuRd', 'PuRd_r', 'Purples', 'Purples_r', 'RdBu', 'RdBu_r', 'RdGy', 'RdGy_r', 'RdPu', 'RdPu_r', 'RdYlBu', 'RdYlBu_r', 'RdYlGn', 'RdYlGn_r', 'Reds', 'Reds_r', 'Set1', 'Set1_r', 'Set2', 'Set2_r', 'Set3', 'Set3_r', 'Spectral', 'Spectral_r', 'Wistia', 'Wistia_r', 'YlGn', 'YlGnBu', 'YlGnBu_r', 'YlGn_r', 'YlOrBr', 'YlOrBr_r', 'YlOrRd', 'YlOrRd_r', 'afmhot', 'afmhot_r', 'autumn', 'autumn_r', 'binary', 'binary_r', 'bone', 'bone_r', 'brg', 'brg_r', 'bwr', 'bwr_r', 'cividis', 'cividis_r', 'cool', 'cool_r', 'coolwarm', 'coolwarm_r', 'copper', 'copper_r', 'crest', 'crest_r', 'cubehelix', 'cubehelix_r', 'flag', 'flag_r', 'flare', 'flare_r', 'gist_earth', 'gist_earth_r', 'gist_gray', 'gist_gray_r', 'gist_heat', 'gist_heat_r', 'gist_ncar', 'gist_ncar_r', 'gist_rainbow', 'gist_rainbow_r', 'gist_stern', 'gist_stern_r', 'gist_yarg', 'gist_yarg_r', 'gnuplot', 'gnuplot2', 'gnuplot2_r', 'gnuplot_r', 'gray', 'gray_r', 'hot', 'hot_r', 'hsv', 'hsv_r', 'icefire', 'icefire_r', 'inferno', 'inferno_r', 'jet', 'jet_r', 'magma', 'magma_r', 'mako', 'mako_r', 'nipy_spectral', 'nipy_spectral_r', 'ocean', 'ocean_r', 'pink', 'pink_r', 'plasma', 'plasma_r', 'prism', 'prism_r', 'rainbow', 'rainbow_r', 'rocket', 'rocket_r', 'seismic', 'seismic_r', 'spring', 'spring_r', 'summer', 'summer_r', 'tab10', 'tab10_r', 'tab20', 'tab20_r', 'tab20b', 'tab20b_r', 'tab20c', 'tab20c_r', 'terrain', 'terrain_r', 'turbo', 'turbo_r', 'twilight', 'twilight_r', 'twilight_shifted', 'twilight_shifted_r', 'viridis', 'viridis_r', 'vlag', 'vlag_r', 'winter', 'winter_r'

.../python3.8/site-packages/matplotlib/_api/__init__.py:126: ValueError
```